contracts/MarginPool.sol

```diff
@@ -3,10 +3,135 @@
+     *      so amounts are scaled down to native token decimals using _calcTransferAmount().
+     *      If _asset equal to WETH address, transfer WETH from Controller address to pool
+     * @param _asset address of asset to transfer
+     * @param _user address of user to transfer assets from
+     * @param _amount amount of token to transfer from _user, scaled to 1e18 of precision
+     * @return true if successful transfer
+     */
+    function transferToPool(
+        address _asset,
+        address _user,
+        uint256 _amount
+    ) external onlyController returns (bool) {
+        require(_amount > 0, "MarginPool: transferToPool amount is below 0");
+
+        // get asset decimals
+        uint8 assetDecimal = ERC20Interface(_asset).decimals();
+        // scale amount
+        uint256 val = _calcTransferAmount(_amount, assetDecimal);
+
+        bool success;
+
+        /// check if asset is WETH or not, if WETH transfer the val from Controller address
+        if (_asset == address(WETH)) {
+            // tranfer WETH from controller to pool
+            success = ERC20Interface(_asset).transferFrom(msg.sender, address(this), val);
+        } else {
+            // transfer val from _user to pool
+            success = ERC20Interface(_asset).transferFrom(_user, address(this), val);
+        }
+
+        return success;
+    }
+
+    /**
+     * @notice transfers asset from pool to user
+     * @dev all tokens are scaled to have 1e18 precision in contracts,
+     *      so amounts are scaled down to native token decimals using _calcTransferAmount()
+     * @param _asset address of asset to transfer
+     * @param _user address of user to transfer assets to
+     * @param _amount amount of token to transfer to _user, scaled to 1e18 of precision
+     * @return true if successful transfer
+     */
+    function transferToUser(
+        address _asset,
+        address payable _user,
+        uint256 _amount
+    ) external onlyController returns (bool) {
+        require(_amount > 0, "MarginPool: transferToUser amount is below 0");
+
+        // get asset decimals
+        uint8 assetDecimal = ERC20Interface(_asset).decimals();
+        // scale amount
+        uint256 val = _calcTransferAmount(_amount, assetDecimal);
+
+        // check if asset is WETH or not, if WETH transfer the val from pool to Controller address
+        if (_asset == address(WETH)) {
+            // unwrap WETH
+            WETH.withdraw(val);
+            // transfer ETH to user
+            _user.transfer(val);
+
+            return true;
+        } else {
+            // transfer asset val from Pool to _user
+            return ERC20Interface(_asset).transfer(_user, val);
+        }
+    }
 
+    /**
+     * @notice Scale _amt
+     * @param _amt amount of tokens
+     * @param _decimals token decimals
+     * @return scaled amount
+     */
+    function _calcTransferAmount(uint256 _amt, uint256 _decimals) internal pure returns (uint256) {
```

Comment:
@aparnakr @antoncoding this function should be here, right ?

Reply:
No I don't think we need this, for transferToUser, we can calculate the amount in controller and call `Pool.transferToUser()`, for transferToPool, we can call it with the amount value in the arguments (un-scaled)

Reply:
cool, I will update this. cc @aleone

Reply:
so @antoncoding the amount in the args params will be by default scaled, in the case of `transferToUser()` ?

Reply:
yeah, in both cases, you don't have to deal with scaling.